plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1176,10 +1180,31 @@ private void executeExpireSnapshots(ConnectorSession session, IcebergTableExecut
                 IcebergSessionProperties.EXPIRE_SNAPSHOTS_MIN_RETENTION);
 
         long expireTimestampMillis = session.getStart().toEpochMilli() - retention.toMillis();
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        List<String> pathsToDelete = new ArrayList<>();
+        Consumer<String> deleteFunction = path -> {
+            pathsToDelete.add(path);
+            if (pathsToDelete.size() == DELETE_BATCH_SIZE) {
+                try {
+                    fileSystem.deleteFiles(pathsToDelete);
+                    pathsToDelete.clear();
+                }
+                catch (IOException e) {
+                    throw new TrinoException(GENERIC_INTERNAL_ERROR, "Failed to delete files during snapshot expiration.", e);
+                }
+            }
+        };
 
         table.expireSnapshots()
                 .expireOlderThan(expireTimestampMillis)
+                .deleteWith(deleteFunction)
```

Comment:
Does the `deleteFunction` need to be thread safe?
Can `table.expireSnapshots()` use threads under the hood?

(some Iceberg APIs do use threads implicitly, not sure about this one today, and not sure whether it can change tomorrow)

Reply:
Normally `table.expireSnapshots()` is single threaded unless you set executor with
```
.executeDeleteWith(executor)
```
I will add an explanatory comment